### PR TITLE
NMS-14548: Provisioning - Config SMNP Community does not support for more than 10 locations

### PR DESCRIPTION
**Pull request by rabzobi**
### All Contributors

* [X ] Have you read our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [X] Have you (electronically) signed the [OpenNMS Contributor Agreement](https://cla-assistant.io/OpenNMS/opennms)?

### Contribution Checklist

* Please [make an issue in the OpenNMS issue tracker](https://issues.opennms.org) if there isn't one already.<br />Once there is an issue, please:
  1. update the title of this PR to be in the format: `${JIRA-ISSUE-NUMBER}: subject of pull request`
  2. update the JIRA link at the bottom of this comment to refer to the real issue number
  3. prefix your commit messages with the issue number, if possible
  4. once you've created this PR, please link to it in a comment in the JIRA issue
  Don't worry if this sounds like a lot, we can help you get things set up properly.
* **If this code is likely to affect the UI, did you name your branch with `-smoke` in it to trigger smoke tests?**
* If this is a new or updated feature, is there documentation for the new behavior?
* If this is new code, are there unit and/or integration tests?
* If this PR targets a `foundation-*` branch, does it try to avoid changing files in `$OPENNMS_HOME/etc/`?

### What's Next?

A PR should be assigned at least 2 reviewers.  If you know that someone would be a good person to review your code, feel free to add them.

If you need help making additions or changes to the documentation related to your changes, please let us know.

In any case, if anything is unclear or you want help getting your PR ready for merge, please don't hesitate to say something in the comments here,
or in [the #opennms-development chat channel](https://chat.opennms.com/opennms/channels/opennms-development).

Once reviewer(s) accept the PR and the branch passes continuous integration, the PR is eligible for merge.

At that time, if you have commit access (are an OpenNMS Group employee or a member of the OGP) you are welcome to merge the PR when you're ready.
Otherwise, a reviewer can merge it for you.

Thanks for taking time to contribute!

### External References

* JIRA (Issue Tracker): http://issues.opennms.org/browse/NMS-14548

### CODE REVIEW

**Review by amay94 — CHANGES_REQUESTED** [review]
Can you add a unit test for this issue to `MonitoringLocationRestServiceIT`?

**Review by RangerRick — DISMISSED** [review]

File: opennms-web-api/src/main/java/org/opennms/web/utils/QueryParameters.java [hunk 1]
```diff
@@ -30,7 +30,7 @@
 
 public class QueryParameters {
 
-    private static final Integer DEFAULT_LIMIT = 10;
+    private static final Integer DEFAULT_LIMIT = 100;
```
Comment (j-white) [inline opennms-web-api/src/main/java/org/opennms/web/utils/QueryParameters.java:33]: Does this have other implications?

File: opennms-webapp-rest/src/main/java/org/opennms/web/rest/v2/MonitoringLocationRestService.java [hunk 2]
```diff
@@ -88,10 +88,10 @@ protected Class<OnmsMonitoringLocation> getQueryBeanClass() {
 
     @Override
     protected CriteriaBuilder getCriteriaBuilder(UriInfo uriInfo) {
-        final CriteriaBuilder builder = new CriteriaBuilder(OnmsMonitoringLocation.class);
+        CriteriaBuilder builder = new CriteriaBuilder(OnmsMonitoringLocation.class);
 
-        // Order by location name by default
-        builder.orderBy("locationName").asc();
+        // Order by location name by default and remove limit on list
+        builder = builder.orderBy("locationName").asc().limit(0);
```
Comment (j-white) [inline opennms-webapp-rest/src/main/java/org/opennms/web/rest/v2/MonitoringLocationRestService.java:94]: Are we always returning the full set of results and ignoring the limit now?

File: opennms-webapp-rest/src/test/java/org/opennms/web/rest/v2/MonitoringLocationRestServiceIT.java [hunk 3]
```diff
@@ -28,10 +28,15 @@
...
 
+import java.util.ArrayList;
 import java.util.Collections;
+import java.util.concurrent.ThreadLocalRandom;
 
 import javax.ws.rs.core.MediaType;
 
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import net.bytebuddy.implementation.bind.annotation.Default;
```
Comment (j-white) [inline opennms-webapp-rest/src/test/java/org/opennms/web/rest/v2/MonitoringLocationRestServiceIT.java:39]: Needed?

File: opennms-webapp-rest/src/test/java/org/opennms/web/rest/v2/MonitoringLocationRestServiceIT.java [hunk 4]
```diff
@@ -164,4 +171,93 @@ public void testEventOnCreationAndDeletion() throws Exception {
...
+            localLocations.add(loc);
+
+            String location = "<location location-name=\"" + loc.locationName + "\" monitoring-area=\""+loc.monitoringArea+"\"/>";
+            sendPost("/monitoringLocations", location, 201, null);
+        }
+
+        // Fetch count and check it against local count
+        String remoteCount = sendRequest(GET, "/monitoringLocations/count", Collections.emptyMap(), 200);
+        LOG.info("testListLimitAndSort: remoteCount="+remoteCount+" localCount="+localLocations.size());
+        assertTrue (localLocations.size() == Integer.parseInt(remoteCount));
```
Comment (j-white) [inline opennms-webapp-rest/src/test/java/org/opennms/web/rest/v2/MonitoringLocationRestServiceIT.java:206]: Avoid `assertTrue` when possible - if it assertion does fail, the error message it gives isn't helpful. This is better written as:
```
assertEquals(localLocations.size(), Integer.parseInt(remoteCount))
```
or
```
assertThat(Integer.parseInt(remoteCount), equalTo(localLocations.size()))
```

**Comment by j-white** [comment]
I think it would make more sense to change the behavior of the caller's side (request all results with a limit of 0) rather than changing the REST API.

File: opennms-web-api/src/main/java/org/opennms/web/utils/QueryParameters.java (hunk 1 above)
Comment (rabzobi) [reply opennms-web-api/src/main/java/org/opennms/web/utils/QueryParameters.java:33]: It affects other drop downs which have the incorrect limit. which aren't mentioned in the original "quick win" ticket, eg "Home / Admin / Configure SNMP by IP". I will do an investigation into other implications.

**Pull request closed without merging**